tests/ui/deriving/deriving-smart-pointer.rs

```diff
@@ -4,6 +4,7 @@
 use std::marker::SmartPointer;
 
 #[derive(SmartPointer)]
+#[repr(transparent)]
 struct MyPointer<'a, #[pointee] T: ?Sized> {
     ptr: &'a T,
 }
```

Comment:
Does it still work if you swap the order?
```rs
#[repr(transparent)]
#[derive(SmartPointer)]
struct MyPointer<'a, #[pointee] T: ?Sized> {
    ptr: &'a T,
}
```

Reply:
Yes, and a test is just now added to certify it.